servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/HostSelector.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright © 2018-2023 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+import io.servicetalk.client.api.LoadBalancedConnection;
+import io.servicetalk.concurrent.api.Single;
+import io.servicetalk.context.api.ContextMap;
+
+import java.util.List;
+import java.util.function.Predicate;
+import javax.annotation.Nullable;
+
+/**
+ * Interface abstracting away the method of host selection.
+ *
+ * This is useful for supporting multiple load balancing algorithms such as round-robin,
```

Comment:
nit: I feel like describing the alternative possibilities, makes it feel that we always provide them out of the box; which is not going to be the case. Consider sth more generic.

Reply:
I just deleted the elaboration for now since we're going to be re-shaping it in a bit.